Backend code

radl97 opened the pull request:
This is a version of the backend where the admin role management is
missing. Coming in next patch. Also the file uploads/downloadis

CassioAI committed:
Backend code

This is a version of the backend where the admin role management is
missing. Coming in next patch. Also the file uploads/downloadis

radl97 commented on API/main.py:
```diff
@@ -0,0 +1,109 @@
+    file.save(upload_folder+"\\"+file.filename)
+    return Response(response="Upload succesful", status=200)
+
+@main.route('/caffs/<caff_id>', methods=['GET'])
```
```suggestion
@main.route('/caffs/<int: caff_id>', methods=['GET'])
```

I've found this here: https://stackoverflow.com/questions/14350920/define-a-route-for-url-ending-with-integer-in-python

radl97 commented on API/main.py:
```diff
@@ -0,0 +1,109 @@
+    }
+    return json_data
+
+@main.route('/caffs/<caff_id>', methods=['DELETE'])
```
```suggestion
@main.route('/caffs/<int: caff_id>', methods=['DELETE'])
```

radl97 commented on API/main.py:
```diff
@@ -0,0 +1,109 @@
+    db.session.commit()
+    return Response(status=200)
+
+@main.route('/caffs/<caff_id>/comments', methods=['POST'])
```
```suggestion
@main.route('/caffs/<int: caff_id>/comments', methods=['POST'])
```

radl97 commented on API/main.py:
```diff
@@ -0,0 +1,109 @@
+    db.session.commit()
+    return Response(status=200)
+
+@main.route('/caffs/<caff_id>/comments/<comment_id>', methods=['DELETE'])
```
```suggestion
@main.route('/caffs/<int: caff_id>/comments/<int: comment_id>', methods=['DELETE'])
```

radl97 commented on API/main.py:
```diff
@@ -0,0 +1,109 @@
+    db.session.commit()
+    return Response(status=200)
+
+@main.route('/caffs/<caff_id>/download', methods=['GET'])
```
```suggestion
@main.route('/caffs/<int: caff_id>/download', methods=['GET'])
```

radl97 commented on API/main.py:
```diff
@@ -0,0 +1,109 @@
+    caff = db.session.query(CAFF).get(caff_id)
+    comment = db.session.query(Comment).get(comment_id)
+    db.session.delete(comment)
+    caff.comments -= 1
```
isn't it `number_of_comments`? Or this is just DB name?

CassioAI replied:
just DB field name:: 
![image](https://user-images.githubusercontent.com/41287236/143836289-c2c55ddf-ac41-4756-b822-f46f88ddb40d.png)

radl97 commented:
Can you commit a requirements.txt please?

radl97 committed:
Move to upload folder; small fixes

radl97 commented on backend/API/__init__.py:
```diff
@@ -0,0 +1,39 @@
+
+# init SQLAlchemy so we can use it later in our models
+db = SQLAlchemy()
+upload_folder ="uploads\\"
```
this kills linux versions (I now have a `uploads\` directory lol)

```suggestion
upload_folder ="uploads" +  os.pathsep
```

radl97 commented on backend/API/main.py:
```diff
@@ -0,0 +1,109 @@
+    new_caff = CAFF(url=upload_folder+"\\"+file.filename, name=file.filename, comments =0, uploader=current_user.id)
+    db.session.add(new_caff)
+    db.session.commit()
+    file.save(upload_folder+"\\"+file.filename)
```
upload_folder already contains the path separator (`\`)

```suggestion
    file.save(upload_folder+file.filename)
```

radl97 commented on backend/API/main.py:
```diff
@@ -0,0 +1,109 @@
+    caff = CAFF.query.filter_by(name=file.filename, uploader=current_user.id).first()
+    if caff or file.filename.split('.')[-1] != "caff":
+        return Response(response='File already uploaded', status=400)
+    new_caff = CAFF(url=upload_folder+"\\"+file.filename, name=file.filename, comments =0, uploader=current_user.id)
```
upload_folder already contains the path separator (`\`)

```suggestion
    new_caff = CAFF(url=upload_folder+file.filename, name=file.filename, comments =0, uploader=current_user.id)
```

CassioAI committed:
Fixing comments

This path fixes the issues brought up in the comments

CassioAI committed:
Fixing comments

This path fixes the issues brought up in the comments

radl97 committed:
Add manager for adding admins

CassioAI committed:
Merge pull request #14 from radl97/backend-admin-manager

Backend admin manager

radl97 committed:
Add CORS

radl97 committed:
Fix requirements.txt

radl97 committed:
Fix missing deps

radl97 committed:
Fix admin role

radl97 committed:
Merge branch 'backend-cors' of github.com:radl97/BiztonsagosGang-2021 into backend

CassioAI committed:
Small fixes and Admin

This patch includes small fixes in retriveing the data from the requests, and tries to introduce admin handling

radl97 committed:
Merge remote-tracking branch 'origin/main' into backend

radl97 committed:
Use Swagger UI for debugging

radl97 committed:
Merge remote-tracking branch 'origin/main' into backend

radl97 committed:
Fix paths for Linux

radl97 committed:
Add error handling (404 error cases)

radl97 committed:
Iterate on download and upload

radl97 committed:
Many fixes

radl97 committed:
Use unauthorized random-generated endpoint for downloads

radl97 committed:
Add modify comment

radl97 committed:
Fix syntax error

radl97 committed:
Fix name not present

CassioAI committed:
Password check

Adding password stragth check to the authentication

CassioAI committed:
Password check

Adding password stragth check to the authentication

radl97 committed:
Fix /caffs/x/download

radl97 committed:
Merge branch 'backend' of github.com:radl97/BiztonsagosGang-2021 into backend

radl97 committed:
Fix

radl97 committed:
Respond with JSON for /signup endpoint

radl97 committed:
Fix /download endpoint
